test/core/pin.js

```diff
@@ -44,14 +45,22 @@ describe('pin', function () {
   let pin
   let repo
 
-  function expectPinned (hash, type, pinned = true) {
+  function expectPinned (hash, type = pinTypes.all, pinned = true) {
```

Comment:
`expectPinned` was not checking if returned `type` match requested one, so I've added double-check for that.